File: src/main/java/org/demoth/aworlds/server2/ActorService.java
package org.demoth.aworlds.server2;

import org.demoth.aworlds.server2.api.LongPropertiesEnum;
import org.demoth.aworlds.server2.model.Location;
import org.demoth.aworlds.server2.model.Player;
import org.springframework.stereotype.Component;

@Component
public class ActorService {
    public Player loadCharacter(String charId) {
        Player player = new Player();
        player.setName(charId);
        player.setType("PLAYER");
        return player;
    }

    public void setLocation(Player character) {
        character.setLocation(new Location());
        character.setLong(LongPropertiesEnum.X, 4L);
        character.setLong(LongPropertiesEnum.Y, 7L);
    }

    public Player createCharacter(String[] params) {
        return null;
    }
}

File: src/main/kotlin/org/demoth/aworlds/server2/ActorService.kt
package org.demoth.aworlds.server2

import org.demoth.aworlds.server2.api.LongPropertiesEnum
import org.demoth.aworlds.server2.model.Location
import org.demoth.aworlds.server2.model.Player
import org.springframework.stereotype.Component

@Component
open class ActorService {
    fun loadCharacter(charId: String): Player {
        val player = Player()
        player.name = charId
        player.type = "PLAYER"
        player.setLong(LongPropertiesEnum.SIGHT_RADIUS, 1L)
        return player
    }

    fun setLocation(character: Player) {
        character.location = Location()
        character.setLong(LongPropertiesEnum.X, 3L)
        character.setLong(LongPropertiesEnum.Y, 2L)
    }
}
